radio/src/gui/colorlcd/model_select.cpp

```diff
@@ -366,41 +367,46 @@ ModelsPageBody::ModelsPageBody(Window *parent, const rect_t &rect) :
 {
   setFlexLayout(LV_FLEX_FLOW_ROW_WRAP, MODEL_CELL_PADDING);
   padRow(MODEL_CELL_PADDING);
-  update();
 }
 
 void ModelsPageBody::selectModel(ModelCell *model)
 {
-  bool modelConnected =
-      TELEMETRY_STREAMING() && !g_eeGeneral.disableRssiPoweroffAlarm;
-  if (modelConnected) {
-    AUDIO_ERROR_MESSAGE(AU_MODEL_STILL_POWERED);
-    if (!confirmationDialog(STR_MODEL_STILL_POWERED, nullptr, false, []() {
-          tmr10ms_t startTime = getTicks();
-          while (!TELEMETRY_STREAMING()) {
-            if (getTicks() - startTime > TELEMETRY_CHECK_DELAY10ms) break;
-          }
-          return !TELEMETRY_STREAMING() || g_eeGeneral.disableRssiPoweroffAlarm;
-        })) {
-      return;  // stop if connected but not confirmed
+  // Don't need to check connection to receiver if re-selecting the active model
+  if (model != modelslist.getCurrentModel()) {
+    bool modelConnected =
+        TELEMETRY_STREAMING() && !g_eeGeneral.disableRssiPoweroffAlarm;
+    if (modelConnected) {
+      AUDIO_ERROR_MESSAGE(AU_MODEL_STILL_POWERED);
```

Comment:
@philmoz I finally found a bug from this PR, which is quite nasty as it can completely lock the radio up. With the Quick Model select option on (it only triggers in the ON state), and you have a telemetry receiver active, when trying to switch model, you will get this warning, but it will keep repeating the message, and the haptic vibration continuously rather than a single time, and the radio will need to be hard powered off even if you do power off the receiver (and the prompts stop). This was with 2.8.1 - `main` does not seem to do this.

Reply:
I think that was fixed in PR #3061 with the changes to the full screen dialog.
One of the commits in that PR mentions fixing the infinite loop issue.

Reply:
Thanks - I'll see if https://github.com/EdgeTX/edgetx/pull/3061/commits/96cfe35071e8e365d767f1daf5e20dc6eadef409 fixes it then and push it in for 2.8.2 if it behaves. :)

Reply:
Now that you mention that PR, that reminds me... would you be able to double check if my memory is correct/why some of the alerts from  `storage/sdcard_yaml.cpp` : `const char * loadRadioSettingsYaml(bool checks)` don't have any touch buttons? I think it's the ones that come up if there aren't any radio settings present.

Reply:
There are three variation for the full screen popup:
- confirmation dialog (with Enter and RTN buttons), generated with confirmationDialog() call
- alert dialog with one action button defined by caller, generated with raiseAlert() and non-empty 3rd param
- alert dialog with no buttons, generated by ALERT macro or raiseAlert() with empty 3rd param

sdcard_yaml.cpp uses the ALERT macro.